### PR TITLE
R/gateway settings

### PR DESCRIPTION
**Pull request by kovrus**

### CODE REVIEW

File: docs/configuration.txt [hunk 1]
```diff
@@ -1125,6 +1125,39 @@ Metadata
...
+  The setting ``gateway.expected_nodes`` defines the number of nodes should be
+  waited for until the cluster state is recovered immediately.
+
+.. _gateway.recover_after_time:
+
+**gateway.recover_after_time**
+  | *Default:*   ``5m``
+  | *Runtime:*  ``no``
+
+  The ``gateway.recover_after_time`` setting defines the time to wait before
```
Comment (mfussenegger) [inline docs/configuration.txt]: does this description here really make sense?

Comment (mfussenegger) [inline docs/configuration.txt]: the descriptions in https://crate.io/docs/en/latest/best_practice/multi_node_setup.html#gateway-configuration make much more sense to me.

**Comment by mfussenegger** [comment]
please check the docs with @chaudum ,  LGTM otherwise

File: docs/configuration.txt [hunk 2]
```diff
@@ -1125,6 +1125,46 @@ Metadata
   Defines how often the cluster collect metadata information
   (e.g. disk usages etc.) if no concrete  event is triggered.
 
+Gateway Settings
```
Comment (chaudum) [inline docs/configuration.txt]: Gateway could be subheading of Metadata, right?

File: docs/configuration.txt [hunk 3]
```diff
@@ -1125,6 +1125,46 @@ Metadata
...
+  changes. This data is stored persistently across full cluster restarts and
+  recovered after nodes are started again.
+
+.. _gateway.expected_nodes:
+
+**gateway.expected_nodes**
+  | *Default:*   ``-1``
+  | *Runtime:*  ``no``
+
+  The setting ``gateway.expected_nodes`` defines the number of nodes should be
```
Comment (chaudum) [inline docs/configuration.txt]: of nodes **that** should ...

File: docs/configuration.txt [hunk 4]
```diff
@@ -1125,6 +1125,46 @@ Metadata
...
+
+**gateway.recover_after_nodes**
+  | *Default:*   ``-1``
+  | *Runtime:*  ``no``
+
+  The ``gateway.recover_after_nodes`` setting defines the number of nodes that
+  need to be started before the cluster state recovery will start. Ideally the
+  value of the setting should be equal to the number of nodes in the cluster,
+  because you only want the cluster state to be recovered once all nodes are
+  started.
```
Comment (chaudum) [inline docs/configuration.txt]: MUST be at least the quorum

File: docs/configuration.txt [hunk 5]
```diff
@@ -1125,6 +1126,49 @@ Metadata
...
+
+**gateway.recover_after_nodes**
+  | *Default:*   ``-1``
+  | *Runtime:*  ``no``
+
+  The ``gateway.recover_after_nodes`` setting defines the number of nodes that
+  need to be started before the cluster state recovery will start. Ideally the
+  value of the setting should be equal to the number of nodes in the cluster,
+  because you only want the cluster state to be recovered once all nodes are
+  started. However, the value must be bigger than the half of the maximum
```
Comment (chaudum) [inline docs/configuration.txt]: ~~maximum~~ expected

**Comment by chaudum** [comment]
pls rebase, otherwise ack

**Pull request merged**